src/robot/result/model.py

```diff
@@ -41,22 +41,25 @@
 from datetime import datetime, timedelta
 from itertools import chain
 from pathlib import Path
-from typing import cast, Mapping, Sequence, Type, Union
+from typing import cast, Generic, Mapping, Sequence, Type, Union, TypeVar
+
 if sys.version_info >= (3, 8):
     from typing import Literal
 
 from robot import model
 from robot.model import (BodyItem, create_fixture, DataDict, Keywords, Tags,
                          SuiteVisitor, TotalStatistics, TotalStatisticsBuilder,
                          TestCases, TestSuites)
-from robot.utils import get_elapsed_time, setter
+from robot.utils import copy_signature, get_elapsed_time, KnownAtRuntime, setter
 
 from .configurer import SuiteConfigurer
 from .messagefilter import MessageFilter
 from .modeldeprecation import deprecated, DeprecatedAttributesMixin
 from .keywordremover import KeywordRemover
 from .suiteteardownfailed import SuiteTeardownFailed, SuiteTeardownFailureHandler
 
+BT = TypeVar('BT', 'IfBranch', 'TryBranch')
+IT = TypeVar('IT', 'ForIteration', 'WhileIteration')
```

Comment:
See earlier comments about `bound` usage and naming. If we'd use `TI` instead of `BT`, then `IT` here could be `FW` or `WF`. Perhaps the latter would be better because the former looks like an abbreviation of `Framework`.